src/Test.UnitTests.Sarif/Baseline/ResultMatching/SarifLogResultMatcherTests.cs

```diff
@@ -387,6 +387,53 @@ public void SarifLogResultMatcher_PreservesPropertiesProperly()
             matchedLog.Runs[0].Results?.Where((r) => { return r.GetProperty("Key") == currentPropertyValue; }).Count().Should().Be(matchedLog.Runs[0].Results.Count);
         }
 
+        [Fact]
+        public void SarifLogResultMatcher_PreservesVersionControlProvenanceProperly()
```

Comment:
>SarifLogResultMatcher_PreservesVersionControlProvenanceProperly [](http://example.com/codeflow?start=20&length=63)

You need to include some null scenarios as well.

Reply:
added null value test cases.